src/components/crossDevice/CrossDeviceLink/index.js

```diff
@@ -142,131 +147,239 @@ class CrossDeviceLinkUI extends Component {
-      alert(`No SMS will be sent, please copy this link ${window.location.origin}`)
+      alert(
+        `No SMS will be sent, please copy this link ${window.location.origin}`
+      )
     }
 
     const { language, sms, token, urls } = this.props
     const url = urls.telephony_url
     const options = {
-      payload: JSON.stringify({to: sms.number, id: this.linkId, language}),
+      payload: JSON.stringify({ to: sms.number, id: this.linkId, language }),
       endpoint: `${url}/v1/cross_device_sms`,
       contentType: 'application/json',
       token: `Bearer ${token}`
     }
-    performHttpReq(options, this.handleResponse , this.handleSMSError)
+    performHttpReq(options, this.handleResponse, this.handleSMSError)
   }
 
-  mobileUrl = ({hosted_sdk_url}) =>
-    // This lets us test the cross device flow locally and on surge.
+  getMobileUrl = () => {
+    const { hosted_sdk_url } = this.props.urls
+    // This lets us test the cross device flow locally and on Surge.
     // We use the same location to test the same bundle as the desktop flow.
-    process.env.MOBILE_URL === "/" ?
+    return process.env.MOBILE_URL === "/" ?
       `${window.location.origin}?link_id=${this.linkId}` :
       `${hosted_sdk_url}/${this.linkId}`
+  }
 
   clearSendLinkClickTimeout() {
     if (this.sendLinkClickTimeoutId) {
       clearTimeout(this.sendLinkClickTimeoutId)
     }
   }
 
+  renderSmsLinkSection = () => {
+    const { translate } = this.props
+    const { sending, validNumber } = this.state
+    const buttonCopyKey = sending ?
+      'cross_device.link.button_copy.status' :
+      'cross_device.link.button_copy.action'
+    const invalidNumber = !validNumber
+    return (
+      <div>
+        <div className={style.smsSection}>
+          <div aria-live="polite" className={style.label}>
+            {translate('cross_device.link.sms_label')}
+          </div>
+          <div className={style.numberInputSection}>
+            <div
+              className={classNames(style.inputContainer, {
+                [style.fieldError]: invalidNumber
+              })}
+            >
+              <PhoneNumberInputLazy
+                {...this.props}
+                clearErrors={this.clearErrors}
+              />
+            </div>
+            <Button
+              ariaLive="polite"
+              ariaBusy={sending}
+              className={classNames(style.btn, { [style.sending]: sending })}
+              variants={['primary']}
+              onClick={this.handleSendSmsLinkClick}
+              disabled={sending || invalidNumber}
+            >
+              {translate(buttonCopyKey)}
+            </Button>
+          </div>
+        </div>
+        <div role="alert" hidden={!invalidNumber}>
+          {invalidNumber && (
+            <div className={style.numberError}>
+              {translate('errors.invalid_number.message')}
+            </div>
+          )}
+        </div>
+      </div>
+    )
+  }
+
+  renderCopyLinkSection = () => {
```

Comment:
This functional component is pretty big. This file is becoming pretty large in size, I would consider putting this component in a different file

Reply:
Yeah, did want to split these out into separate components but didn't get around to it. 

Have split out the Copy Link view as that is pretty straightforward. Send SMS needs a bit more time and care for the refactor. Will leave that as it is for now.